[skip-ci] Move RHEL branch to TMT

lsm5 opened the pull request:
<!--
Thanks for sending a pull request!

Please make sure you've read and understood our contributing guidelines
(https://github.com/containers/buildah/blob/main/CONTRIBUTING.md) as well as ensuring
that all your commits are signed with `git commit -s`.
-->

#### What type of PR is this?

<!--
Please label this pull request according to what type of issue you are
addressing, especially if this is a release targeted pull request.

Uncomment only one `/kind <>` line, hit enter to put that in a new line, and
remove leading whitespace from that line:
-->


> /kind other

#### What this PR does / why we need it:

#### How to verify it

#### Which issue(s) this PR fixes:

<!--
Automatically closes linked issue when PR is merged.
Uncomment the following comment block and include the issue
number or None on one line.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`, or `None`.
-->

None

#### Special notes for your reviewer:

None

#### Does this PR introduce a user-facing change?

<!--
If no, just write `None` in the release-note block below. If yes, a release note
is required: Enter your extended release note in the block below. If the PR
requires additional action from users switching to the new release, include the
string "action required".

For more information on release notes please follow the kubernetes model:
https://git.k8s.io/community/contributors/guide/release-notes.md
-->

```release-note
None
```

lsm5 committed:
Cirrus: remove config

Signed-off-by: Lokesh Mandvekar <lsm5@fedoraproject.org>

lsm5 commented:
The validate test uses a bunch of Cirrus variables that I'm not sure can be transferred easily to TMT.

How bad would it be to skip validate test on RHEL branches? I'm guessing there'll be no community contributions directly to RHEL branches, and quite a few layers of QE.

lsm5 commented:
/packit test

lsm5 commented:
We'll need branch protection setup to block on the TMT jobs (likewise for every new branch that gets used in RHEL).

Once the PR looks good, I can add that if everyone agrees.

lsm5 commented:
I'll say ignore the epel-9-aarch64 `in-progress` for now  as it seems like a final status reporting issue. I've reported the issue to packit. Also, we're not testing on aarch64 envs yet, so that shouldn't be a blocker.

lsm5 commented:
@nalind @flouthoc @TomSweeneyRedHat @Luap99 @mheon PTAL

flouthoc reviewed (approved):
LGTM one nit above.

Luap99 commented:
This also removes a basic things like checking if the tree (vendor) dir is clean which I think is generally important.
I don't know about the importance of the other removed test cases.

And than of course our old discussion around how we deal with merge protection. I think if we do this than we need to turn of the total success check on the older branches which means we cannot really on the merge no merging to early and users must manually ensure they only merge with passing CI.
I am not involved in backports of this repo so I don't care much so I guess this is mostly relevant to @nalind and @TomSweeneyRedHat.

lsm5 commented:
> This also removes a basic things like checking if the tree (vendor) dir is clean which I think is generally important. I don't know about the importance of the other removed test cases.

I'll add the vendoring test., that one should be no problem. Thanks.

lsm5 committed:
Move RHEL branch to TMT

Advantages:
- Test RHEL branches on on actual RHEL
- Reduce our GCE/AWS usage
- Easier to switch git host (RE: CNCF)

Note: This setup does not do rpm builds, binaries are built from source
itself using Makefile. There's no clean and reliable way to fetch actual
RHEL spec files using packit, and CentOS Stream spec files can't be
depended on as they move at a faster pace and can be out of sync with
RHEL release spec files.

The lint test has been disabled for now.

Fixes: RUN-3105

Signed-off-by: Lokesh Mandvekar <lsm5@fedoraproject.org>

lsm5 commented:
`make lint` is failing on CI and also on my local env. Let me know if anyone got a fix for that, or else I'd like to skip it.

```
go build -o build/golangci-lint ./vendor/github.com/golangci/golangci-lint/cmd/golangci-lint
make[1]: Leaving directory '/var/ARTIFACTS/work-lintt206ahvk/plans/lint/tree/tests/tools'
./tests/tools/build/golangci-lint run 
run_common.go:384:3: missing type in composite literal (typecheck)
		{Name: string(specs.CgroupNamespace), Host: cfg.CgroupNS() == "host"},
		^
run_common.go:385:3: missing type in composite literal (typecheck)
		{Name: string(specs.IPCNamespace), Host: cfg.IPCNS() == "host"},
		^
run_common.go:386:3: missing type in composite literal (typecheck)
		{Name: string(specs.MountNamespace), Host: false},
		^
run_common.go:379:2: declared and not used: cfg (typecheck)
	cfg, err := config.Default()
	^
cmd/buildah/from.go:236:48: iopts.Isolation undefined (type fromReply has no field or method Isolation) (typecheck)
	isolation, err := parse.IsolationOption(iopts.Isolation)
	                                              ^
cmd/buildah/from.go:256:69: iopts.CapAdd undefined (type fromReply has no field or method CapAdd) (typecheck)
	capabilities, err := defaultContainerConfig.Capabilities("", iopts.CapAdd, iopts.CapDrop)
	                                                                   ^
cmd/buildah/from.go:263:44: iopts.DecryptionKeys undefined (type fromReply has no field or method DecryptionKeys) (typecheck)
	decConfig, err := cli.DecryptConfig(iopts.DecryptionKeys)
	                                          ^
cmd/buildah/from.go:272:32: iopts.GroupAdd undefined (type fromReply has no field or method GroupAdd) (typecheck)
		GroupAdd:              iopts.GroupAdd,
		                             ^
cmd/buildah/from.go:280:32: iopts.CNIPlugInPath undefined (type fromReply has no field or method CNIPlugInPath) (typecheck)
		CNIPluginPath:         iopts.CNIPlugInPath,
		                             ^
cmd/buildah/from.go:281:32: iopts.CNIConfigDir undefined (type fromReply has no field or method CNIConfigDir) (typecheck)
		CNIConfigDir:          iopts.CNIConfigDir,
		                             ^
cmd/buildah/from.go:286:32: iopts.BlobCache undefined (type fromReply has no field or method BlobCache) (typecheck)
		BlobDirectory:         iopts.BlobCache,
		                             ^
cmd/buildah/from.go:287:32: iopts.Devices undefined (type fromReply has no field or method Devices) (typecheck)
		DeviceSpecs:           iopts.Devices,
		                             ^
cmd/buildah/from.go:288:32: iopts.Retry undefined (type fromReply has no field or method Retry) (typecheck)
		MaxPullRetries:        iopts.Retry,
		                             ^
cmd/buildah/from.go:290:32: iopts.CDIConfigDir undefined (type fromReply has no field or method CDIConfigDir) (typecheck)
		CDIConfigDir:          iopts.CDIConfigDir,
		                             ^
cmd/buildah/from.go:293:11: iopts.RetryDelay undefined (type fromReply has no field or method RetryDelay) (typecheck)
	if iopts.RetryDelay != "" {
	         ^
cmd/buildah/from.go:294:58: iopts.RetryDelay undefined (type fromReply has no field or method RetryDelay) (typecheck)
		options.PullRetryDelay, err = time.ParseDuration(iopts.RetryDelay)
		                                                       ^
cmd/buildah/from.go:296:86: iopts.RetryDelay undefined (type fromReply has no field or method RetryDelay) (typecheck)
			return fmt.Errorf("unable to parse value provided %q as --retry-delay: %w", iopts.RetryDelay, err)
			                                                                                  ^
cmd/buildah/run.go:171:27: iopts.CNIPlugInPath undefined (type runInputOptions has no field or method CNIPlugInPath) (typecheck)
		CNIPluginPath:    iopts.CNIPlugInPath,
		                        ^
cmd/buildah/run.go:172:27: iopts.CNIConfigDir undefined (type runInputOptions has no field or method CNIConfigDir) (typecheck)
		CNIConfigDir:     iopts.CNIConfigDir,
		                        ^
level=error msg="Timeout exceeded: try increasing it by passing --timeout option"
make: *** [Makefile:193: lint] Error 4
make: Leaving directory '/var/ARTIFACTS/work-lintt206ahvk/plans/lint/tree'
Shared connection to 10.31.40.144 closed.

```

Luap99 commented on internal/mkcw/embed/entrypoint_amd64.gz:
I assume this should not be changed

lsm5 replied:
fixed.

Luap99 reviewed (commented):
typecheck errors happen when golangci is unable to compile the program, a common issue is a missing library or something like that
But I guess the linter on a backport is not really a critical thing to run so I am fine if this doesn't work

flouthoc reviewed (approved):
LGTM
/lgtm
/approve

flouthoc commented:
@lsm5 `[testing-farm:RHEL-9.6.0-Nightly-x86_64]` is failing is this expected ?

TomSweeneyRedHat commented:
LGTM

TomSweeneyRedHat commented:
@lsm5 This branch just transitioned to the Sustaining team, and this version of Buildah is on RHEL 9.6 and 10.0, which came out this past May.
